pyanaconda/modules/payloads/payload/dnf/dnf_manager.py

```diff
@@ -706,14 +704,22 @@ def _run_transaction(base, display):
 
         try:
             base.do_transaction(display)
-            exit_reason = "DNF done"
+            error_items = False
+            for tsi in base.transaction:
+                if tsi.state == libdnf.transaction.TransactionItemState_ERROR:
+                    log.error("The transaction item %s ended in error state.", tsi)
+                    error_items = True
+            if error_items:
```

Comment:
I would move this section to a new method that would return True or False. It would help with readability of this part.

Reply:
I was thinking you would, agreed:)

Reply:
As _run_transaction is a static method I am putting the new function in payload/dnf/utils.py (rather than making _run_transaction and the new method class methods).